src/Components/Address/NewAddressForm.tsx

```diff
@@ -0,0 +1,261 @@
+import * as Yup from "yup"
+import { Button, Column, GridColumns, Input, Message } from "@artsy/palette"
+import { Formik, Form, FormikConfig, useFormikContext } from "formik"
+import { FC, useEffect } from "react"
+import { CountrySelect } from "Components/CountrySelect"
+
+export interface AddressFormValues {
+  name?: string
+  country?: string
+  addressLine1?: string
+  city?: string
+  region?: string
+  postalCode?: string
+  addressLine2?: string
+}
+
+export type AddressChangeHandler = (
+  address: AddressFormValues,
+  key: keyof AddressFormValues
+) => void
+
+export const EMPTY_ADDRESS = {
+  name: "",
+  country: "",
+  addressLine1: "",
+  addressLine2: "",
+  city: "",
+  postalCode: "",
+  region: "",
+}
+
+export const DEFAULT_VALIDATION_SCHEMA = Yup.object().shape({
+  name: Yup.string().required("Name is required"),
+  country: Yup.string().required("Country is required"),
+  addressLine1: Yup.string().required("Address is required"),
+  city: Yup.string().required("City is required"),
+  region: Yup.string().required("Region is required"),
+  postalCode: Yup.string().required("Postal Code is required"),
+})
+
+// For debugging
```

Comment:
Stuff that is secondary to the primary concern of the component should be moved to the bottom of the file or to another file